CMakeLists.txt

```diff
@@ -496,13 +496,9 @@ endif ()
 
 enable_testing() # Enable for tests without binary
 
-option(ENABLE_EXTERNAL_OPENSSL "This option is insecure and not recommended for any occasions. If it is enabled, it allows building with alternative OpenSSL library. By default, ClickHouse is using BoringSSL, which is better. Do not use this option." OFF)
```

Comment:
OpenSSL 3.0 has quite a bad track record in terms of security in comparison to BoringSSL. It means the comment makes sense.

Reply:
Yes, however, it is much easier to delegate responsibility to the OS vendor for maintaining and owning the the security posture. For example, https://access.redhat.com/errata/RHSA-2022:7288 .  Having the customer do a package upgrade is often must easier and it is the responsibility of redhat in this case (rather than clickhouse).

We also have the issue of FIPS validation only applying to packaged versions of SSL, meaning we cannot use the source code directly, we have to link either statically or dynamically to a FIPS validated version. (Yes, there is one for boringSSL but its not typically maintained as well)

Lastly, we are porting to s390x ( z platform). The project has so far not been willing to maintain a s390x port.